### PR TITLE
Force disable older versions of tony's enhancements add-on with 2023.3

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:
None

### Summary of the issue:
Tony's enhancements version 1.51 and older prevents any speech from happening when WASAPI is enabled.
This is due to the add-on relying on private API functions.
Now that WASAPI is enabled by default, upgrading to 2023.3 resulting in silence should be avoided if possible.

### Description of user facing changes
When updating to NVDA 2023.3, if tony's enhancement add-on is installed it will trigger the typical UX warning for upgrading to an API breaking release, where incompatible add-ons are force disabled.

### Description of development approach
Force known versions of problematic add-ons to be considered disabled with 2023.3.
This code should be cleaned up in 2024.1.

### Testing strategy:
- [x] Test installing all add-ons in the add-on store to confirm if other add-ons are affected.
- [x] Test upgrading to the try build from 2023.2

### Known issues with pull request:
None

### Change log entries:
Refer to diff

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] Security precautions taken.

### CODE REVIEW

**Comment by leonardder** [comment]
While I appreciate this action, wouldn't it be better to disable WASAPI by default in 2023.3 and enable it in 2024.1 instead? People who really need WASAPI can still enable it.

**Comment by cary-rowen** [comment]
Hi @seanbudd 

I release a new version of tony's enhancements with Tony's permission. This version is compatible with 2023.2 and 2023.3.

The add-on is currently available on the NV Access add-on store.

Cheers
Cary

**Comment by cary-rowen** [comment]
But the overall goal of this PR still makes sense, and there may be other add-ons that require such a solution.

**Comment by amirsol81** [comment]
@leonardder Arguments can be made for and against this, but, IMO, it would be better to make WASAPI the default in 2023.3. It's become a lot more stable with the recent alphas.

**Comment by CyrilleB79** [comment]
Now that Tony's add-on seems to be fixed. Should this PR still be merged?

If yes, I think a strong clarification / explanation from NV Access should be done. Else, this gives the impression that:
* add-on authors can use private methods (not in the API)
* if things break NVDA in the future due to the use of private methods, NV Access will provide a fix, not to fix the add-on's capability but at least to avoid that NVDA be out of order

IMO if the add-on is very popular and NV Access wants to help, it would be a better choice to submit a PR to the add-on (as for NVDA remote) than putting fixes in NVDA's code base.

At last, regarding this PR, should only the add-ons in the add-on store be looked at? Or should we mention and report add-ons  that are not (yet) in the store but that are popular in the various communities?

**Comment by seanbudd** [comment]
>
While I appreciate this action, wouldn't it be better to disable WASAPI by default in 2023.3 and enable it in 2024.1 instead? People who really need WASAPI can still enable it.

NVDA core will be prioritised over add-ons. Especially in this case, where the add-on breakage is due to relying on private API functions. Incompatible add-ons themselves being disabled is more typical of the NVDA UX as well.

>Now that Tony's add-on seems to be fixed. Should this PR still be merged?

We think so, yes. This case when updating with an older version of this add-on installed is still really bad.
I've removed some of the special handling (in regards to the warning message to disable WASAPI) as an updated version of the add-on is now available.
While it is extremely frustrating when add-ons relying on the private functions hamper NVDA core development, it would be irresponsible of us to release 2023.3 knowing that many users will be upgrading to silence without an easy path to fix it.

Per section 15 of [the product vision](https://github.com/nvaccess/nvda/blob/master/projectDocs/product_vision.md):
"NVDA Protects users from installing an add-on that is incompatible with the current version of NVDA, allowing the user to override this protection if they understand the risk.".

So I would suggest your second point "if things break NVDA in the future due to the use of private methods, NV Access will provide a fix, not to fix the add-on's capability but at least to avoid that NVDA be out of order" is closer to our current policy.

>IMO if the add-on is very popular and NV Access wants to help, it would be a better choice to submit a PR to the add-on (as for NVDA remote) than putting fixes in NVDA's code base.

Fixing community made add-ons is also beyond our product vision at stands, rather we try and safeguard NVDA core to prevent such issues. Reasoning for this is fairly clear in this example, with no simple fix to the problem requiring the author to remove features from the add-on.

>At last, regarding this PR, should only the add-ons in the add-on store be looked at? Or should we mention and report add-ons that are not (yet) in the store but that are popular in the various communities?

We've only tested add-ons in the store. Testing every add-on outside of the store is beyond our capacity. If other add-on issues are reported before the RC.

File: source/_addonStore/models/version.py [hunk 1]
```diff
@@ -62,7 +62,8 @@ def enableCompatibilityOverride(self):
 		"""
 		from addonHandler import AddonStateCategory, state
 		overiddenAddons = state[AddonStateCategory.OVERRIDE_COMPATIBILITY]
-		assert self.name not in overiddenAddons and self.canOverrideCompatibility
+		assert self.name not in overiddenAddons, f"{self.name}, {overiddenAddons}"
```
Comment (michaelDCurran) [inline source/_addonStore/models/version.py:65]: The single assertion was an 'and' of two conditions. The two assertions replacing this are therefore an 'or'. Is this definitely what you meant?

Comment (seanbudd) [reply source/_addonStore/models/version.py:65]: I don't think this is true, both assertions must pass and therefore it's an `and`? If one of these is false, the other assertion would still fail.

Comment (michaelDCurran) [reply source/_addonStore/models/version.py:65]: Yep, you're right, sorry about that.

**Review by michaelDCurran — APPROVED** [review]

**Comment by XLTechie** [comment]
Why is overridden being spelled wrong in this code? (overidden)

**Pull request merged**